test/tests/functional/pbs_verify_log_output.py

```diff
@@ -113,31 +111,28 @@ def test_if_info_add(self):
         """
         Test for interface info presence in log files
         """
-        names = "["
-        for name in [x[0] for x in self.all_interfaces()]:
-            names += "(" + name + ")"
-        names += "]"
-
-        log_val = "[(ipv4)(ipv6)(ipv4/ipv6)] [(interface: )]" + names
-
-        rv1 = self.scheduler.log_match(
-            log_val,
-            regexp=True,
-            starttime=self.server.ctime,
-            max_attempts=5,
-            interval=2)
-        rv2 = self.server.log_match(
-            log_val,
-            regexp=True,
-            starttime=self.server.ctime,
-            max_attempts=5,
-            interval=2)
-        rv3 = self.mom.log_match(
-            log_val,
-            regexp=True,
-            starttime=self.server.ctime,
-            max_attempts=5,
-            interval=2)
-        self.assertTrue(rv1)
-        self.assertTrue(rv2)
-        self.assertTrue(rv3)
+        interfaceGenerator = self.all_interfaces()
+        for ifname in interfaceGenerator:
+            name = "[(" + ifname + ")]"
+            log_val = "[( interface: )]" + name
+            rv1 = self.scheduler.log_match(
+                log_val,
+                regexp=True,
+                starttime=self.server.ctime,
+                max_attempts=5,
+                interval=2)
+            rv2 = self.server.log_match(
+                log_val,
+                regexp=True,
+                starttime=self.server.ctime,
+                max_attempts=5,
+                interval=2)
+            rv3 = self.mom.log_match(
+                log_val,
+                regexp=True,
+                starttime=self.server.ctime,
+                max_attempts=5,
+                interval=2)
+            self.assertTrue(rv1)
+            self.assertTrue(rv2)
+            self.assertTrue(rv3)
```

Comment:
Put the asserts after each individual log_match().  If all three aren't matched, you'll wait 10s each (30s) just to finally realize the test failed.  You could have easily realized the test is going to fail after the first.